tools/trap/ramdumpParser.py

```diff
@@ -620,39 +626,63 @@ def find_crash_point(log_file):
 
             #It displays the debug symbols corresponding to all the addresses in the application text address range
-            print('-------------------------- DEBUG SYMBOLS IN "{0}" TEXT RANGE -------------------------'.format(app_name[app_idx]))
+            print('\n-------------------------- DEBUG SYMBOLS IN "{0}" TEXT RANGE -------------------------'.format(app_name[app_idx]))
             os.system("python ../debug/debugsymbolviewer.py " + app_name[app_idx] + ".map " + log_file + " " + hex(int(hex(g_stext_app[app_idx]), 16)) + " 0")
 
-    # Check for pc & lr values in kernel text address range
-    if (address1 >= hex(g_stext_flash) and address1 < hex(g_etext_flash)) or (address1 >= hex(g_stext_ram) and address1 < hex(g_etext_ram)):
-        # If yes, print the crash point using addr2line
-        f = os.popen('arm-none-eabi-addr2line -a -f -e' + elf_path + ' ' + hex(lr_value))
-        result = f.read()
-        if '??' not in result and '$d' not in result:
-            print('Kernel Crash point is as follows:')
-            print('[ Caller - return address (LR) - of the function which has caused the crash ]')
-            format_output(result)
-    elif (address2 >= hex(g_stext_flash) and address2 < hex(g_etext_flash)) or (address2 >= hex(g_stext_ram) and address2 < hex(g_etext_ram)):
-        f = os.popen('arm-none-eabi-addr2line -a -f -e' + elf_path + ' ' + hex(pc_value))
-        result = f.read()
-        if '??' not in result and '$d' not in result:
-            print('Kernel Crash point is as follows:')
-            print('[ Current location (PC) of assert ]')
-            print(' - Exact crash point might be -4 or -8 bytes from the PC.')
-            format_output(result)
-    elif (not is_app_crash):
-        print('PC & LR values not in any text range, no probable crash point detected')
-        print('-----------------------------------------------------------------------------------------\n')
+    # Check for pc & lr values in application text address range
+    if (not is_app_crash):
+        # Check for pc & lr values in kernel text address range
+        if (address1 >= hex(g_stext_flash) and address1 < hex(g_etext_flash)) or (address1 >= hex(g_stext_ram) and address1 < hex(g_etext_ram)):
+            # If yes, print the crash point using addr2line
+            f = os.popen('arm-none-eabi-addr2line -a -f -e' + elf_path + ' ' + hex(lr_value))
+            result = f.read()
+            if '??' not in result and '$d' not in result:
+                is_kernel_crash = 1
+                print('Kernel Crash point is as follows:')
+                print('[ Caller - return address (LR) - of the function which has caused the crash ]\n')
+                format_output(result)
+        if (address2 >= hex(g_stext_flash) and address2 < hex(g_etext_flash)) or (address2 >= hex(g_stext_ram) and address2 < hex(g_etext_ram)):
+            f = os.popen('arm-none-eabi-addr2line -a -f -e' + elf_path + ' ' + hex(pc_value))
```

Comment:
Can you please pass elf to the function find_crash_point(log_file) instead of using elf_path? Currently, this only works if the kernel elf file is named as tinyara.